data_structures/src/data_request.rs

```diff
@@ -367,14 +377,19 @@ pub fn create_tally(
         outputs.push(vt_output_change);
     }
 
-    TallyTransaction::new(dr_pointer, consensus_tally, outputs)
+    // FIXME: make this factory return a Result, as this line could easily fail and right now we are
+    //  ignoring failures and putting empty vectors (which will fail when decoded)
+    let tally_bytes = Vec::try_from(report).unwrap_or_default();
```

Comment:
Why not fail immediately?

```suggestion
    let tally_bytes = Vec::try_from(report).unwrap();
```